### PR TITLE
mupdf: update to 1.24.2

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

Update

Also needed due to `gumbo-parser` update in https://github.com/macports/macports-ports/pull/24110

###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.6
Xcode 3.2

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: graphics/mupdf/Portfile [hunk 1]
```diff
@@ -20,49 +19,71 @@ long_description    MuPDF is a lightweight PDF and XPS viewer and \
...
-checksums           rmd160  63d39c7cfc70ac08beed28d8da6e581bbf09ae6f \
-                    sha256  deb0347a3f9ce0967611d479a8a07b57e2c730b0aaf09dce28ee41abb9ac4838 \
-                    size    55146083
+checksums           rmd160  e19bad3ca13762fe5b238093d62e5ce292b69df4 \
+                    sha256  7fbced6d6024608661d773b93990119db7eb2483f0d3e28855eadef9ca9a8686 \
+                    size    55313640
 
-depends_build       port:pkgconfig
+depends_build-append \
+                    bin:pkg-config:pkgconfig
```
Comment (reneeotten) [inline graphics/mupdf/Portfile:42]: as @ryandesign pointed out in a different PR; it seems like one should use "path/bin:....". I presume that that syntax will only check for things in "${prefix}/bin"

**Review by reneeotten — CHANGES_REQUESTED** [review]

**Comment by pmetzger** [comment]
You will need to resolve the conflict.

File: graphics/mupdf/Portfile (hunk 1 above)
Comment (barracuda156) [reply graphics/mupdf/Portfile:42]: Yes, this should be changed, will do it today.

**Comment by barracuda156** [comment]
@pmetzger @reneeotten Hopefully all resolved now.

**Pull request merged**